File: Java/Enemy.java
import java.util.Random;

public class Enemy {
	public double enHealth;
	public double enArmor;
	public int enPotions;
	public String enType;
	public String enName;
	public boolean isBoss;
	public double damageTaken;
	public boolean Alive;
	public double totalEnHealth;
	public Random randomizer;
	private int typeRandomizer;
	public double damage;
	public double accuracy;
	private int level;
	private int proToLevel = 0;
	
	// Specific Enemy
	public Enemy(double Health, double Armor, int Potions, String Type, boolean Boss ) {
		enHealth = Health;
		enArmor = Armor;
		enPotions = Potions;
		enType = Type;
		enName = Type;
		isBoss = Boss;
		totalEnHealth = Health;
		
		}
	
	// Random Enemy
	public Enemy() {
		randomizer = new Random();
		enHealth = (double)randomizer.nextInt(200 + (100 * level));
		enArmor = (double)randomizer.nextInt(25 + (15 * level));
		enPotions = randomizer.nextInt(3);
		typeRandomizer = randomizer.nextInt(2);
		
		if(typeRandomizer == 0) {
			enType = "Archer";
		}
		if(typeRandomizer == 1) {
			enType = "Bandit";
		}
		if (typeRandomizer == 2) {
			enType = "RogueKnight";
		}
		
		enName = enType;
		isBoss = false;
		totalEnHealth = enHealth;
	}
		
	public double getEnHealth() {
		return enHealth;
	}
	public double getTotalEnHealth() {
		return totalEnHealth;
	}
	
	public double getEnArmor() {
		return enArmor;
	}
	
	public boolean isBoss() {
		return isBoss;
	}
	
	public int getEnPotions() {
		return enPotions;
	}
	
	public String getName() {
		return enName;
	}
	
	public String getEnType() {
		return enType;
		
	}
	public boolean isAlive() {
			return Alive;}
	
	public void isDefeated(User Warrior) {
		if (this.enHealth <= 0) {
			
			Alive = false;
			System.out.println(this.getName() + " was defeated!");
			Warrior.deafetedEnemiesCounter(this); 
			proToLevel++;
			if (proToLevel >= 10) {
				level++;
				proToLevel = 0;
			}
			
		}
	}
	public int getLevel() {
		return level;
	}
	
	
	public double getDamage() {
		return damage;
	}
	
	
	
	public void takeDamage(double damage, User warrior) {
		damageTaken = damage - this.getEnArmor();
		if (damageTaken <= 0) {
			System.out.println("Their armor was to Strong" + damageTaken);
		}
		if (damageTaken > 0) {
		enHealth -= damageTaken;
		System.out.println(this.getName() + " was hit for " + damageTaken + " damage!" );
		if(this.enHealth <= 0) {
			isDefeated(warrior);
			}
		}			
	}
	
	
	public void usePotion() {
		this.enHealth += 200;
		if(this.enHealth > this.getTotalEnHealth()) {
			this.enHealth = this.getTotalEnHealth();
		}
		this.enPotions--;
		
			
			}
	public void attack(User warrior) {
		System.out.println(this.getName() + " tries to attack");
		double accCheck = randomizer.nextDouble();
		if (accCheck <= accuracy) {	
			warrior.takeDamage(getDamage());
			
			} 
		else {
			System.out.println(this.getName() + " missed!");
			}
		}

	//public Enemy generateEnemy(String genType) {
	//	String enType = genType;
	//	return Enemy((double)randomizer.nextInt(1000), (double)randomizer.nextInt(1000),randomizer.nextInt(3),
				//	enType, false);
		//
		
		
	

	
	}

	
		
	

	
	

	




	

File: JavaScript/Enemy.js


class Enemy {
	//proToLevel = 0;
	
	// Specific Enemy
    /*constructor(Health, Armor, Potions, Type, Boss ) {
		enHealth = Health;
		enArmor = Armor;
		enPotions = Potions;
		enType = Type;
		enName = Type;
		isBoss = Boss;
		totalEnHealth = Health;
		
	}*/

	
	// Random Enemy
	constructor(level, isBoss) {

		
		this.enPotions = Math.floor(Math.random() * 4);
		this.typeRandomizer = Math.floor(Math.random() * 3);
		

		if(this.typeRandomizer == 0) {
			this.enType = "Archer";
			this.enHealth = Math.floor(Math.random() * 100) + (50 * level);
			this.enArmor = Math.floor(Math.random() * 10) + (15 * level);
			this.damage = Math.floor(Math.random() * 50) + (20 * level);
			this.accuracy = Math.random() + 0.3;
		}
		if(this.typeRandomizer == 1) {
			this.enType = "Bandit";
			this.enHealth = Math.floor(Math.random() * 150) + (50 * level);
			this.enArmor = Math.floor(Math.random() * 15) + (15 * level);
			this.damage = Math.floor(Math.random() * 40) + (20 * level);
			this.accuracy = Math.random() + 0.35;
		}
		if (this.typeRandomizer == 2) {
			this.enType = "RogueKnight";
			this.enHealth = Math.floor(Math.random() * 200) + (50 * level);
			this.enArmor = Math.floor(Math.random() * 20) + (15 * level);
			this.damage = Math.floor(Math.random() * 35) + (20 * level);
			this.accuracy = Math.random() + 0.40;
		}
		
		this.enName = this.enType;
		if (isBoss) {
			this.enHealth += Math.floor(Math.random() * 300) + (200 * level);
			this.enArmor += 20;
			this.damage += Math.floor(Math.random() * 60) + (10 * level);

		}
		
		this.totalEnHealth = this.enHealth;
	}
	
	getEnHealth() {
		return enHealth;
	}
	getTotalEnHealth() {
		return totalEnHealth;
	}
	
	getEnArmor() {
		return enArmor;
	}
	
	isBoss() {
		return isBoss;
	}
	
	getEnPotions() {
		return enPotions;
	}
	
	getName() {
		return enName;
	}
	
	getEnType() {
		return enType;
		
	}
	isAlive() {
			return Alive;}
	
	isDefeated(Warrior) {
		if (this.enHealth <= 0) {
			
			Alive = false;
			System.out.println(this.getName() + " was defeated!");
			Warrior.deafetedEnemiesCounter(this); 
			proToLevel++;
			if (proToLevel >= 10) {
				level++;
				proToLevel = 0;
			}
			
		}
	}
	getLevel() {
		return level;
	}
	
	
	getDamage() {
		return damage;
	}
	
	
	
	takeDamage(damage, warrior) {
		var damageTaken = damage - this.enArmor;
		/*if (damageTaken <= 0) {
			System.out.println("Their armor was to Strong" + damageTaken);
		}*/
		if (damageTaken > 0) {
		this.enHealth -= damageTaken;
		
		if(this.enHealth <= 0) {
			this.enHealth = 0
			//isDefeated(warrior);
			}
		}
		if (damageTaken <= 0) {
			return 0
		}
		else{
			return damageTaken	
		}		
	}
	
	
	usePotion() {
		this.enHealth += 200;
		if(this.enHealth > this.getTotalEnHealth()) {
			this.enHealth = this.getTotalEnHealth();
		}
		this.enPotions--;
		
			
			}
	attack(warrior) {
		System.out.println(this.getName() + " tries to attack");
		accCheck = randomizer.nextDouble();
		if (accCheck <= accuracy) {	
			warrior.takeDamage(getDamage());
			
			} 
		else {
			System.out.println(this.getName() + " missed!");
			}
		}

	//public Enemy generateEnemy(String genType) {
	//	String enType = genType;
	//	return Enemy((double)randomizer.nextInt(1000), (double)randomizer.nextInt(1000),randomizer.nextInt(3),
				//	enType, false);
		//
		
		
	

	
}

	
		
	
	
